db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2742,6 +2742,10 @@ Status DBImpl::EnableAutoCompaction(
       s = status;
     }
   }
+  {
+    InstrumentedMutexLock l(&mutex_);
+    MaybeScheduleFlushOrCompaction();
+  }
```

Comment:
Should we SchedulePendingCompaction() before MaybeScheduleFlushOrCompaction()? If `disable_auto_compactions` had been "true" for that CFD, the CFD will not be added to the queue as https://github.com/facebook/rocksdb/blob/01f2edd145e318228936b89834d058060ed600c6/db/db_impl/db_impl_compaction_flush.cc#L3005. `MaybeScheduleFlushOrCompaction()` only schedule thread to pull from that queue https://github.com/facebook/rocksdb/blob/01f2edd145e318228936b89834d058060ed600c6/db/db_impl/db_impl_compaction_flush.cc#L3587 but is meaningless if the CFD is not on the queue.

Reply:
Also, the scheduling is only needed when the SetOptions() succeeds.